packages/headless/src/controllers/commerce/search-box/headless-search-box.test.ts

```diff
@@ -0,0 +1,272 @@
+      updateQuerySetQuery({id: id, query: ''})
+    );
+  });
+
+  it(`when calling clear
+      should dispatch a clearQuerySuggest action`, () => {
+    searchBox.clear();
+    expect(engine.actions).toContainEqual(clearQuerySuggest({id}));
+  });
+
+  describe('#showSuggestions', () => {
+    it('dispatches fetchQuerySuggestions', async () => {
+      searchBox.showSuggestions();
+
+      const action = engine.actions.find(
+        (a) => a.type === fetchQuerySuggestions.pending.type
+      );
+      expect(action).toEqual(
+        fetchQuerySuggestions.pending(action!.meta.requestId, {id})
+      );
+    });
+  });
+
+  describe('#selectSuggestion', () => {
+    it('dispatches a selectQuerySuggestion action', () => {
+      const value = 'i like this expression';
+      searchBox.selectSuggestion(value);
+
+      expect(engine.actions).toContainEqual(
+        selectQuerySuggestion({id, expression: value})
+      );
+    });
+
+    it('dispatches executeSearch', () => {
+      const suggestion = 'a';
+      searchBox.selectSuggestion(suggestion);
+
+      expect(engine.findAsyncAction(executeSearch.pending)).toBeTruthy();
+    });
+  });
+
+  describe('when calling submit', () => {
+    it('it deselects all facets', () => {
+      searchBox.submit();
+
+      expect(engine.actions).toContainEqual(deselectAllBreadcrumbs());
+    });
+
+    it('allows autoSelection', () => {
+      searchBox.submit();
+
+      expect(engine.actions).toContainEqual(
+        updateFacetAutoSelection({allow: true})
+      );
+    });
+
+    it('autoSelection should be allowed after deselecting facets', () => {
+      searchBox.submit();
+
+      const deselectAllBreadcrumbsIndex = engine.actions.findIndex(
+        (action) => action.type === deselectAllBreadcrumbs.type
+      );
+      const updateFacetAutoSelectionIndex = engine.actions.findIndex(
+        (action) => action.type === updateFacetAutoSelection.type
+      );
+      expect(deselectAllBreadcrumbsIndex).toBeLessThanOrEqual(
+        updateFacetAutoSelectionIndex
+      );
+    });
+
+    it('dispatches updateQuery with the correct parameters', () => {
+      const expectedQuery = state.querySet[id];
+      searchBox.submit();
+
+      const action = updateQuery({
+        query: expectedQuery,
+      });
+
+      expect(engine.actions).toContainEqual(action);
+    });
+
+    it('updates the page to the first one', () => {
+      searchBox.submit();
+      expect(engine.actions).toContainEqual(selectPage(1));
+    });
+
+    it('it dispatches an executeSearch action', () => {
+      searchBox.submit();
+
+      const action = engine.actions.find(
+        (a) => a.type === executeSearch.pending.type
+      );
+      expect(action).toBeTruthy();
+    });
+
+    it('it dispatches a clear suggestions action', () => {
+      searchBox.submit();
+      expect(engine.actions).toContainEqual(clearQuerySuggest({id}));
+    });
+  });
+
+  it(`when querySuggest #isLoading state is true,
```

Comment:
We have an earlier test that's called "should return the right state". I think what we should do is have a describe for the #state and put these two tests under it.

Reply:
Done in https://github.com/coveo/ui-kit/pull/3461/commits/5da347b537ec53f1f414c2f9991fb2f6a1525a34